### PR TITLE
feat(app): #1248 added comments for compodocs

### LINKED ISSUE #1248: Add inline docs to existing code for compodoc
@webkhushboo suggested using CompoDoc. Some places there are no docs, some single line docs and some full docblocs, we should be **consistent**, so we will use full docblocs everywhere.

Example from the docs

```typescript
/**
 * @example
 * This is a good example
 * processTarget('yo')
 *
 * @param {string} target  The target to process see {@link Todo}
 * @returns The processed target number
 */
function processTarget(target:string): number;
```

https://compodoc.app/guides/jsdoc-tags.html (also see `routing`)

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1248 

### Notes

A summary of what was achieved in this PR

- [ ] Added comments so that we can generate documentation using compodocs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,3 +150,7 @@ The format is `<type>(<scope>): #<issue-no> <subject>`, examples...
 1. `set GOOGLE_APPLICATION_CREDENTIALS=path\to\key.json`
 2. `firebase setup:emulators:firestore`
 3. `firebase emulators:start`
+
+## Generate documentation
+1. cd web
+2. npm run docs
```
Comment (eddiejaoude) [inline README.md]: Better to have the commands inline code blocks

1. `cd web`
2. `npm run docs`

File: README.md [hunk 2]
```diff
@@ -150,3 +150,7 @@ The format is `<type>(<scope>): #<issue-no> <subject>`, examples...
 1. `set GOOGLE_APPLICATION_CREDENTIALS=path\to\key.json`
 2. `firebase setup:emulators:firestore`
 3. `firebase emulators:start`
+
+## Generate documentation
```
Comment (eddiejaoude) [inline README.md]: Having a blank line after `headers` is consistent with the rest of the file and easier to read

File: web/package.json [hunk 3]
```diff
@@ -46,6 +47,7 @@
     "@angular/platform-browser-dynamic": "7.2.15",
     "@angular/platform-server": "7.2.15",
     "@angular/router": "7.2.15",
+    "@compodoc/compodoc": "^1.1.10",
```
Comment (eddiejaoude) [inline web/package.json]: Should be a `devDependencies`, not need for the deployed version

File: web/src/app/core/resolvers/edit-project.resolver.ts [hunk 4]
```diff
@@ -12,12 +13,22 @@ import { IProject, ProjectModel } from '@shared/models/index.model';
 })
 export class EditProjectResolver implements Resolve<IProject> {
 
+  /**
+   * Life cycle method
+   * @param authService AuthenticationService instance
+   * @param projectService ProjectService
```
Comment (eddiejaoude) [inline web/src/app/core/resolvers/edit-project.resolver.ts]: This one does not have `instance` after it, but I don't think it is needed, it does not add value to the docs. Sorry but I think it is best to remove it from everywhere

File: web/src/app/core/resolvers/pings.resolver.ts [hunk 5]
```diff
@@ -6,22 +6,29 @@ import { ActivatedRouteSnapshot, Resolve, Router } from '@angular/router';
 import { of, Observable } from 'rxjs';
 import { catchError, take } from 'rxjs/operators';
 
-// Third party modules
+// Dashboard hub model and services
```
Comment (eddiejaoude) [inline web/src/app/core/resolvers/pings.resolver.ts]: It is written `DashboardHub` - I have tried to fix when I see them

File: web/src/app/core/services/project.service.ts [hunk 6]
```diff
@@ -70,7 +85,10 @@ export class ProjectService {
       );
   }
 
-  // This function delete the project via uid
+  /**
+   * Delete the project using project uid
+   * @param uid uid of project to be deleted
```
Comment (eddiejaoude) [inline web/src/app/core/services/project.service.ts]: In the compodocs example they include the type..

```typescript
/**
 * @example
 * This is a good example
 * processTarget('yo')
 *
 * @param {string} target  The target to process see {@link Todo}
 * @returns The processed target number
 */
function processTarget(target:string):number;
```

File: web/src/app/core/services/repository.service.ts [hunk 7]
```diff
@@ -9,25 +10,41 @@ import { RepositoriesModel, RepositoryModel } from '@shared/models/index.model';
...
     private afs: AngularFirestore,
     private fns: AngularFireFunctions,
     private authService: AuthenticationService,
     private activityService: ActivityService
   ) { }
 
-  // Forces refresh of users repositories
+  /**
+   * Forces refresh of users repositories
+   * @reutnrs Observable
```
Comment (eddiejaoude) [inline web/src/app/core/services/repository.service.ts]: Typo? `reutnrs`

File: web/src/app/core/services/monitor.service.ts [hunk 8]
```diff
@@ -81,11 +95,23 @@ export class MonitorService {
...
   public deletePingsByMonitor(projectUid: string, monitorUid: string): Observable<boolean> {
     const callable: any = this.fns.httpsCallable('deletePingsByMonitor');
     return callable({ projectUid, monitorUid });
   }
 
+  /**
+   * Send monitor ping manually using cloud function
+   * @param projectUid uid of project
+   * @param monitorUid uid of monitor
+   * @param type project type
```
Comment (eddiejaoude) [inline web/src/app/core/services/monitor.service.ts]: Missing `return`

**Comment by eddiejaoude** [comment]
Looks good. Left a few comments but not for every time...

- Missing `returns` in docbloc 
- Missing `type` in docbloc

File: README.md (hunk 1 above)
Comment (webkhushboo) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (webkhushboo) [reply README.md]: done

File: web/package.json (hunk 3 above)
Comment (webkhushboo) [reply web/package.json]: done

File: web/src/app/core/resolvers/edit-project.resolver.ts (hunk 4 above)
Comment (webkhushboo) [reply web/src/app/core/resolvers/edit-project.resolver.ts]: Removed from all places

File: web/src/app/core/resolvers/pings.resolver.ts (hunk 5 above)
Comment (webkhushboo) [reply web/src/app/core/resolvers/pings.resolver.ts]: okay done

File: web/src/app/core/services/project.service.ts (hunk 6 above)
Comment (webkhushboo) [reply web/src/app/core/services/project.service.ts]: But it looks like redundant to me to add the type in comment .

File: web/src/app/core/services/repository.service.ts (hunk 7 above)
Comment (webkhushboo) [reply web/src/app/core/services/repository.service.ts]: Oppps. Fixed

File: web/src/app/core/services/monitor.service.ts (hunk 8 above)
Comment (webkhushboo) [reply web/src/app/core/services/monitor.service.ts]: Fixed

File: web/src/app/core/services/monitor.service.ts [hunk 9]
```diff
@@ -65,6 +74,12 @@ export class MonitorService {
       );
   }
 
+  /**
+   * Delete the monitor from project
+   * @param projectUid uid of project
+   * @param monitorUid uid of monitor to be deleted
+   * @returns the observable
```
Comment (eddiejaoude) [inline web/src/app/core/services/monitor.service.ts]: Not consistent with the `returns` below, this is a description and below is the `type`.

Many places do not the return, we need to always be consistent.

I think we discussed to remove the `returns` as the documentation adds it automatically? Is the correct?

File: web/src/app/core/services/monitor.service.ts [hunk 10]
```diff
@@ -82,12 +97,25 @@ export class MonitorService {
       );
   }
 
+  /**
+   * Call cloud function for deleting pings by monitor
+   * @param projectUid uid of project
+   * @param monitorUid uid of monitor
+   * @returns Observable
```
Comment (eddiejaoude) [inline web/src/app/core/services/monitor.service.ts]: See comment above

File: web/src/app/core/services/monitor.service.ts (hunk 9 above)
Comment (webkhushboo) [reply web/src/app/core/services/monitor.service.ts]: removed the @returns from all places.

File: web/src/app/core/services/monitor.service.ts (hunk 10 above)
Comment (webkhushboo) [reply web/src/app/core/services/monitor.service.ts]: done

**Pull request merged**